govcd/vdcComputePolicy.go

```diff
@@ -0,0 +1,246 @@
+func (org *AdminOrg) GetAllVdcComputePolicies(queryParameters url.Values) ([]*VdcComputePolicy, error) {
+	return getAllVdcComputePolicies(org.client, queryParameters)
+}
+
+// GetAllVdcComputePolicies retrieves all VDC compute policies using OpenAPI endpoint. Query parameters can be supplied to perform additional
+// filtering
+func (org *Org) GetAllVdcComputePolicies(queryParameters url.Values) ([]*VdcComputePolicy, error) {
+	return getAllVdcComputePolicies(org.client, queryParameters)
+}
+
+// getAllVdcComputePolicies retrieves all VDC compute policies using OpenAPI endpoint. Query parameters can be supplied to perform additional
+// filtering
+func getAllVdcComputePolicies(client *Client, queryParameters url.Values) ([]*VdcComputePolicy, error) {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointVdcComputePolicies
+	minimumApiVersion, err := client.checkOpenApiEndpointCompatibility(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	responses := []*types.VdcComputePolicy{{}}
+
+	err = client.OpenApiGetAllItems(minimumApiVersion, urlRef, queryParameters, &responses)
+	if err != nil {
+		return nil, err
+	}
+
+	var wrappedVcdComputePolicies []*VdcComputePolicy
+	for _, response := range responses {
+		wrappedVcdComputePolicy := &VdcComputePolicy{
+			client:           client,
+			VdcComputePolicy: response,
+		}
+		wrappedVcdComputePolicies = append(wrappedVcdComputePolicies, wrappedVcdComputePolicy)
+	}
+
+	return wrappedVcdComputePolicies, nil
+}
+
+// CreateVdcComputePolicy creates a new VDC Compute Policy using OpenAPI endpoint
+func (org *AdminOrg) CreateVdcComputePolicy(newVdcComputePolicy *types.VdcComputePolicy) (*VdcComputePolicy, error) {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointVdcComputePolicies
+	minimumApiVersion, err := org.client.checkOpenApiEndpointCompatibility(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := org.client.OpenApiBuildEndpoint(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	returnVdcComputePolicy := &VdcComputePolicy{
+		VdcComputePolicy: &types.VdcComputePolicy{},
+		client:           org.client,
+	}
+
+	err = org.client.OpenApiPostItem(minimumApiVersion, urlRef, nil, newVdcComputePolicy, returnVdcComputePolicy.VdcComputePolicy)
+	if err != nil {
+		return nil, fmt.Errorf("error creating VDC compute policy: %s", err)
+	}
+
+	return returnVdcComputePolicy, nil
+}
+
+// Update updates existing VDC compute policy
+func (vdcComputePolicy *VdcComputePolicy) Update() (*VdcComputePolicy, error) {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointVdcComputePolicies
+	minimumApiVersion, err := vdcComputePolicy.client.checkOpenApiEndpointCompatibility(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	if vdcComputePolicy.VdcComputePolicy.ID == "" {
+		return nil, fmt.Errorf("cannot update VDC compute policy without id")
```

Comment:
Nit: could we use capital `ID` in the messages? It's easier to scan with eyes when you have a lot of log lines.
```suggestion
		return nil, fmt.Errorf("cannot update VDC compute policy without ID")
```

Reply:
changed